src/app/app-components/courses/courses-list/courses-list.component.ts

```diff
@@ -15,48 +18,40 @@ import { map, switchMap } from 'rxjs/operators';
 export class CoursesListComponent implements OnInit {
   @Input() set filterTitle(filterTitle: string) {
     if (filterTitle) {
-      this.courses$ = this.coursesService.list({ textFragment: filterTitle });
+      this.store.dispatch(CoursesActions.listCourses({ textFragment: filterTitle }));
     }
   }
-  courses$: Observable<Course[]>;
+  courses$: Observable<Course[]> = this.store.select(getCoursesStatus);
   private readonly pageSize = 5;
   private pageIndex = 0;
 
-  constructor(private readonly coursesService: CoursesService) {
-    this.courses$ = this.loadByPage(this.previousPage(), this.nextPage())
-      .pipe(map((courses: Course[]) => new OrderByPipe().transform(courses)));
-  }
+  constructor(
+    private readonly store: Store<AppState>,
+  ) { }
 
-  ngOnInit() { }
+  ngOnInit() {
+    this.loadByPage(this.getPage(), this.nextPage());
+  }
 
-  onDeleteItem(idCourse: string) {
+  onDeleteItem(courseId: string) {
     if (confirm('Do you really want to delete this course?')) {
-      // ToDo: Implement deleteById in BE.
-      this.courses$ =
-        this.coursesService.delete(idCourse).pipe(switchMap(() => {
-          this.pageIndex = 0;
-          return this.loadByPage(this.previousPage(), this.nextPage())
-            .pipe(map((courses: Course[]) => new OrderByPipe().transform(courses)));
-        }));
+      this.store.dispatch(CoursesActions.deleteCourse({ courseId }));
     }
   }
 
   onLoadMore() {
-    const result$ = this.loadByPage(this.previousPage(), this.nextPage());
-    this.courses$ = zip(this.courses$, result$)
-      .pipe(map(([a, b]) => [...a, ...b]))
-      .pipe(map((courses: Course[]) => new OrderByPipe().transform(courses)));
+    this.loadByPage(this.getPage(), this.nextPage());
   }
 
-  private loadByPage(start: string, count: string): Observable<Course[]> {
-    return this.coursesService.list({ start, count });
+  private loadByPage(start: string, count: string) {
+    this.store.dispatch(CoursesActions.listCourses({ start, count }));
   }
 
-  private previousPage(): string {
-    return String(this.pageIndex * this.pageSize);
+  private getPage(): string {
+    return String(this.pageSize * this.pageIndex++);
```

Comment:
Just as curiosity: the Google JavaScript Style Guide forbid constructors like `String` or `Boolean`. So, here we could use `(this.pageSize * this.pageIndex++).toString()` instead.

Reply:
Thanks for the info, I'll change it before submitting.